DMCompiler/DM/Builders/DMExpressionBuilder.cs

```diff
@@ -645,8 +645,13 @@ private DMExpression BuildScopeIdentifier(DMASTScopeIdentifier scopeIdentifier,
             // This is the same behaviour as in BYOND, but BYOND simply raises an undefined var error.
             // We want to give end users an explanation at least.
             if (scopeMode is Normal && ctx.Proc != null)
-                return BadExpression(WarningCode.BadExpression, identifier.Location,
-                    "Use of \"type::\" and \"parent_type::\" outside of a context is forbidden");
+                if (ctx.Proc.GetLocalVariable(identifier.Identifier) != null) {
+                    // actually - it's referring to a local variable named "type" or "parent_type"... just do the usual thing
+                    expression = BuildExpression(scopeIdentifier.Expression, inferredPath);
+                } else {
+                    return BadExpression(WarningCode.BadExpression, identifier.Location,
+                            "Use of \"type::\" and \"parent_type::\" outside of a context is forbidden");
```

Comment:
```suggestion
                            "Use of \"type::\" and \"parent_type::\" outside of an object proc definition is forbidden");
```
make the error message a little clearer maybe?

Reply:
@amylizzle  I'm still learning DM but since this fails to compile (byond):

```
/datum/foo
	var/price = 60
	better
		proc/test_proc() 
			price = parent_type::price + 40
```
with:
```
error: parent_type::price: undefined var
error: ::: bad var or proc() after :: operator
```

but no errors here:

```
/datum/foo
	var/price = 60
	better
		price = parent_type::price + 40
```

I'm inclined to have the error be `Use of "type::" or "parent_type::" inside a proc definition requires a local variable with the name "type" or "parent_type"` or something along those lines.

Reply:
huh, I guess I misunderstood what that does? Yeah that's a better message then!